platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ConfigurationMigratable.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+/**
+ * Base contract for all migratables related to system configuration. These are necessary for
```

Comment:
We should try to keep these comments as generic as possible. For instance, we could write something like
> Base interface that must be implemented by all bundles or features that need to export and/or import system related settings during system migration. The information exported must allow the new system to have the same configuration as the original system. Only bundle or feature specific settings need be handled. All configurations stored in OSGi's {@code ConfigurationAdmin} will automatically be migrated and do not need to be managed by implementors of this class.